gusto/transport_equation.py

```diff
@@ -181,6 +183,13 @@ def advection_term(self, q):
                             dot(self.ubar('+'), self.n('+'))*q('+'))
                       + inner(self.test('-'),
                               dot(self.ubar('-'), self.n('-'))*q('-')))*self.dS
+
+        if self.outflow:
+            if self.ibp is not None:
+                L += self.test*self.un*q*self.ds
+            else:
+                raise ValueError("ibp=None is not compatible with outflow option")
```

Comment:
I think I would prefer the error checking to happen in `__init__`, then you can just write:
```python
if self.outflow:
    L += ...
```
here.